### ISSUE TITLE
[IndexTable] selectable prop being set to false has no effect

### ISSUE THREAD

**Issue by gching** [Shopify/polaris-react#4076]
## Issue summary

On `IndexTable`, the boolean prop `selectable` is exposed to enable / disable selectability on `IndexTable`. However, setting `selectable` to `false` does not disable selectability.

## Expected behavior

By setting `selectable` to `false` on `IndexTable`, items should not be selectable in `IndexTable`

## Actual behavior

Setting `selectable` to false allows for items to be selected.

## Reduced test case

Sample `IndexTable` example:

```js
import React from "react";
import {
  TextStyle,
  Card,
  IndexTable,
  useIndexResourceState
} from "@shopify/polaris";

export default function SimpleIndexTableExample() {
  const customers = [
    {
      id: "3411",
      url: "customers/341",
      name: "Mae Jemison",
      location: "Decatur, USA",
      orders: 20,
      amountSpent: "$2,400"
    },
    {
      id: "2561",
      url: "customers/256",
      name: "Ellen Ochoa",
      location: "Los Angeles, USA",
      orders: 30,
      amountSpent: "$140"
    }
  ];
  const resourceName = {
    singular: "customer",
    plural: "customers"
  };

  const {
    selectedResources,
    allResourcesSelected,
    handleSelectionChange
  } = useIndexResourceState(customers);

  const rowMarkup = customers.map(
    ({ id, name, location, orders, amountSpent }, index) => (
      <IndexTable.Row
        id={id}
        key={id}
        selected={selectedResources.includes(id)}
        position={index}
      >
        <IndexTable.Cell>
          <TextStyle variation="strong">{name}</TextStyle>
        </IndexTable.Cell>
        <IndexTable.Cell>{location}</IndexTable.Cell>
        <IndexTable.Cell>{orders}</IndexTable.Cell>
        <IndexTable.Cell>{amountSpent}</IndexTable.Cell>
      </IndexTable.Row>
    )
  );

  return (
    <Card>
      <IndexTable
        selectable={false}
        resourceName={resourceName}
        itemCount={customers.length}
        selectedItemsCount={
          allResourcesSelected ? "All" : selectedResources.length
        }
        onSelectionChange={handleSelectionChange}
        headings={[
          { title: "Name" },
          { title: "Location" },
          { title: "Order count" },
          { title: "Amount spent" }
        ]}
      >
        {rowMarkup}
      </IndexTable>
    </Card>
  );
}
```

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 6.2.0
- Browser: Chrome
- Device: Desktop
- Operating System: Ubuntu

**Comment by jt274** [comment]
I will confirm this. I was just about to come here to open this same issue.

**Comment by mdgozza** [comment]
I can also confirm this behavior and was just coming here to post the same exact issue as well.

Any combination of properties set in order to get the inner components to derive `selectable` as false still do not result in an IndexTable without checkboxes.

This leads me to wonder if it is even possible to have an IndexTable without checkboxes. I sure hope so because this component would be perfect for me without the checkboxes!

Thanks.

**Comment by jt274** [comment]
@mdgozza I agree, it is needed.

I've tried to fork the repo and implement the functionality myself. It certainly appears to not be implemented. However, I will admit I'm new to forking and building someone else's repo and ran into too many issues even trying to install the dependencies and build and test the darn thing.

**Comment by Chompas** [comment]
I'm facing the same problem. According to [Polaris documentation for Index Table ](https://polaris.shopify.com/components/lists-and-tables/index-table):

> The main job of an index table is to help merchants get an at-a-glance of the objects to perform actions or navigate to a full-page representation of it.

I'm trying to use it to navidate to a full-page representation of my row but I don't understand how to accomplish this at the moment since the only action possible seems to be the checkbox.

**Comment by jt274** [comment]
> @Chompas The typical behavior is to have a link to a full-page representation of a data item (for example an order) from one of the index table columns.

Even if you place a link though, the row gets checked when you click the link.

**Comment by gching** [comment]
@jt274 @Chompas How I do navigation is I add the prop `data-primary-link` to the child anchor component (if you are using Next.js, you would have to wrap it with `Link` from `next/link`) inside one of the `IndexTable.Cell`. For example, this is what I kind of have to make navigation work when a user clicks on a row - 

```js
<IndexTable.Cell>
  /** This is from next/link if you are using Next.js **/
  <Link href="YOUR URL" passHref>
      <a data-primary-link>YOUR LINK</a>
  </Link>
</IndexTable.Cell>
```

I found this by looking through the source + tests of `polaris-react`. Of course, it isn't clearly explained in the documentation, so I would suggest you make an issue about the documentation regarding this feature.

**Comment by igor-devx** [comment]
Hi guys,

I faced with the same issue. I add the button inside the cell, but when I click on the button, it checked the row. Please, fix this bug ASAP

**Comment by BlackDragon24** [comment]
> Hi guys,
> 
> I faced with the same issue. I add the button inside the cell, but when I click on the button, it checked the row. Please, fix this bug ASAP

A little late, not sure if your issue is solved.

I added a Polaris Button in the Index Cell and that works.

```javascript
  const rowMarkup = customers.map(
    ({ id, name, location, orders, amountSpent }, index) => (
      <IndexTable.Row
        id={id}
        key={id}
        selected={selectedResources.includes(id)}
        position={index}  
        onClick={() => alert("Hi")}
      >
        <IndexTable.Cell>
          <TextStyle variation="strong">{name}</TextStyle>
        </IndexTable.Cell>
        <IndexTable.Cell>
          <Button onClick={() => alert("Edited")}>
            Edit
          </Button>
        </IndexTable.Cell>
        <IndexTable.Cell>{location}</IndexTable.Cell>
        <IndexTable.Cell>{orders}</IndexTable.Cell>
        <IndexTable.Cell>{amountSpent}</IndexTable.Cell>
      </IndexTable.Row>
    )
  );
```
Unfortunately, what I wanted was to just click anywhere on the row to do what I want, but onClick on IndexTable.Row has no effect. Want the same effect as Resource List.

Button works well though and doesn't check checkbox

I have my IndexTable with bulkactions and promoted bulkactions

**Comment by gching** [comment]
@BlackDragon24 Refer to my comment at https://github.com/Shopify/polaris-react/issues/4076#issuecomment-853487489 - This way lets me click anywhere on the row purely just for navigation - not sure if this helps with your case

**Comment by BlackDragon24** [comment]
> @BlackDragon24 Refer to my comment at [#4076 (comment)](https://github.com/Shopify/polaris-react/issues/4076#issuecomment-853487489) - This way lets me click anywhere on the row purely just for navigation - not sure if this helps with your case

Yeah I tried it, I have embedded app. I don’t want a separate cell that states my link. I tried wrapping the entire IndexTable.Row and that didn’t work.

Clicking the link does work for Cell, but because my app is embedded, navigation still thinks I’m on the same page as the url doesn’t change.

**Comment by daninkc** [comment]
A little late to the party, but maybe this helps someone. I've been struggling with customization for index tables and found a few viable options:
-For clicking an item with custom functionalities (i.e. a TextField, a DatePicker, a button) _without_ selecting/unselecting the row, you can wrap the element inside the <td> with a div that has property onClick={(e) => e.stopPropagation()}. For example:
```
<IndexTable.Row>
            <IndexTable.Cell>
              <div onClick={(e) => e.stopPropagation()}>
                            {Your interactive element here...}
              </div>
            </IndexTable.Cell>
</IndexTable.Row>
```
-For creating custom functionalities for the whole row, for example, making some rows unselectable or changing the click functionality, you can trick polaris into thinking you're using a row by changing the IndexTable.Row to:
` <tr className="Polaris-IndexTable__TableRow"></tr>`
It doesn't show a DOM nesting warning, you can do whatever you want with the tr and you still get the polaris custom styles, up to the pointer. The only downsides are that you don't get a checkbox, so you have to add an empty <td> or add the checkbox manually; and that if polaris changes classnames, you'll have to change these as well. But it can be a nice simple wrapper.

None of these are ideal but they really do work nicely.

Hope it helps somebody :)
